### PR TITLE
config: Add CMake printf variable and C library variables to the generated CMake module

### PR DESCRIPTION
**Pull request by hugueskamba**
### Description

`MBED_TARGET_SUPPORTED_C_LIBS`, `MBED_PRINTF_LIB` and `MBED_C_LIB` are provided
in the generated configuration CMake module to allow the toolchain to be
configured for the selected printf library and C library.

### Test Coverage

- [X]  This change is covered by existing or additional automated tests.
- [ ]  Manual testing has been performed (and evidence provided) as automated testing was not feasible.
- [ ]  Additional tests are not required for this change (e.g. documentation update).

Reviewers
@Patater @rwalton-arm @0xc0170

### CODE REVIEW

**Review by 0xc0170 — APPROVED** [review]

**Comment by rwalton-arm** [comment]
Can you make the PR on a branch off this repo and not a fork? `mbed-tools-ci-scripts` doesn't support forks at the moment, unfortunately...

File: src/mbed_tools/build/_internal/templates/mbed_config.tmpl [hunk 1]
```diff
@@ -7,6 +7,8 @@
 set(MBED_TOOLCHAIN "{{toolchain_name}}" CACHE STRING "")
 set(MBED_TARGET "{{target_name}}" CACHE STRING "")
 set(MBED_CPU_CORE "{{core}}" CACHE STRING "")
+set(MBED_C_LIB "{{c_lib}}" CACHE STRING "")
```
Comment (Patater) [inline src/mbed_tools/build/_internal/templates/mbed_config.tmpl:10]: Should this be `{{clib}}`? All other c_lib or clib references use `clib` (without the underscore).

Will the existing tests catch this?

**Review by Patater — CHANGES_REQUESTED** [review]

File: src/mbed_tools/build/_internal/templates/mbed_config.tmpl (hunk 1 above)
Comment (hugueskamba) [reply src/mbed_tools/build/_internal/templates/mbed_config.tmpl:10]: That should stay like that. It's the others that need to change. Thanks.

Comment (hugueskamba) [reply src/mbed_tools/build/_internal/templates/mbed_config.tmpl:10]: [Fixed](https://github.com/ARMmbed/mbed-tools/compare/c9ae0ed71dd621ec31cb6e3ed8ef8e3229144c36..295bdcd295136fd731251a66fc4091ec1c1d23ee).

**Comment by Patater** [comment]
@hugueskamba
CI won't be able to pass until this PR is closed and a new PR is raised from a branch. Let me know if you don't have write permissions to push a branch to make the new PR from.

**Pull request closed without merging**